### DISCUSSION TITLE
Why do signal check every tick?

### DISCUSSION THREAD

**Discussion by Ohan17** [bitbrain/beehave#63] [Q&A]
In the demo project I see that the signal is checked then connect, disconnect every frame. Can you explain why you end up with this approach? Also have you consider having an internal function to initialize, connect all the needed signals for BeehaveNode class? Something like *ready(actor, blackboard)* that is called from the root down similar to tick but only once.
![image](https://user-images.githubusercontent.com/66206021/205658208-e5b9932d-16d2-4c36-98a8-2d22f47a77b4.png)

**Comment by bitbrain** [comment]
Hello - You are probably referring to this code? https://github.com/bitbrain/godot-tutorials/blob/464a0d7bae41967fa4bd20837665df1a21488a22/behaviour-tree/demo/ai/actions/OakTreeAction.gd#L12

> Can you explain why you end up with this approach? 

I remember writing this code to "make it work" and there is no advantage in doing it this way. Surely, there must be even more efficient ways to do these sorts of checks.

**Reply by Ohan17** [reply]
Thank you for the clarification. This leads to second part of my question as a feature request, is it feasible to have an init function with access to *actor* and *blackboard* as agruments to do these type of works since I have not come up with any alternative.

**Reply by bitbrain** [reply]
That might be a good idea! I am curious what you will come up with.
